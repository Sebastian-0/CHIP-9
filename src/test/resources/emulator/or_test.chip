# Test simple ors
ldi A, 85
ldi B, 170
ldi C, 15
ldi D, 240
ldi E, 1
ldi H, 13
ldi L, 24

or A
or B
or C
or D
or E
or H
or L

sout A
sout B
sout C
sout D
sout E
sout H
sout L

ldx HL, 0x1010
ldi (HL), 7
or (HL)

sout (HL)

# Test flags
ldi A, 6
ldi B, 3
ldi C, 128
ldi D, 0

# No flags
or B
sout FLAGS

# Negative flag
or C
sout FLAGS

# Zero flag
ldi A, 0
or D
sout FLAGS

hcf