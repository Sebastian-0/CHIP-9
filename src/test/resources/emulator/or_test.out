85
255
95
245
85
93
93
87

0
64
128